slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -252,14 +252,12 @@ trait MySQLProfile extends JdbcProfile { profile =>
   }
 
   class UpsertBuilder(ins: Insert) extends super.UpsertBuilder(ins) {
+    // Method kept for binary compatibility. Otherwise unused
```

Comment:
If the next release is v4.0.0 and we don't need to keep binary compatibility, should it be dropped?

Reply:
I don't know if the next release is 4.0.0. Would it make sense to keep bincompat, and open a separate PR that cleans it up more and breaks bincompat? I'm not sure if there's a separate branch or how that will work.

Reply:
Also, can it be marked deprecated and/or access modifiers added?

Reply:
Added deprecated 440d393 .

I'll make a PR for cleaning after merging this. I didn't add an access modifier because it would break binary compatibility.